### PR TITLE
UPSTREAM: 17061: Unnecessary updates to ResourceQuota when doing UPDATE

### PR DESCRIPTION
**Pull request by derekwaynecarr**
https://github.com/kubernetes/kubernetes/pull/17061

Fixes https://github.com/openshift/origin/issues/5827

cc @liggitt please review based on our discussion.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
[test]

File: Godeps/_workspace/src/k8s.io/kubernetes/plugin/pkg/admission/resourcequota/admission.go [hunk 1]
```diff
@@ -168,6 +168,12 @@ func (q *quota) Admit(a admission.Attributes) (err error) {
 // Return true if the usage must be recorded prior to admitting the new resource
 // Return an error if the operation should not pass admission control
 func IncrementUsage(a admission.Attributes, status *api.ResourceQuotaStatus, client client.Interface) (bool, error) {
+	// on update, the only resource that can modify the value of a quota is pods
+	// so if your not a pod, we exit quickly
+	if a.GetOperation() == admission.Update && a.GetResource() != "pods" {
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/plugin/pkg/admission/resourcequota/admission.go]: are there pod subresources we should ignore as well?

Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/plugin/pkg/admission/resourcequota/admission.go]: nm, missed the ignore earlier

**Comment by liggitt** [comment]
approved based on severity discussed in blocker call
LGTM, [merge]

**Pull request merged**